### ISSUE TITLE
Run Travis CI Tests Against Multiple k8s Versions

### ISSUE THREAD

**Issue by seanmalloy** [KohlsTechnology/eunomia#256]
**Is your feature request related to a problem? Please describe.**

The problem is that the automated e2e tests for eunomia that are run through Travis CI are only executed against a single k8s version. The user of eunomia are more than likely running on a variety of k8s versions. Also, right now it is difficult know if updating eunomia dependencies will break eunomia(Go version, operator-sdk version, updating/adding Go modules).

**Describe the solution you'd like**

1. Update minikube to the latest version
2. Create a Travis CI build matrix to run e2e tests in parallel against multiple k8s versions
3. Use k8s versions 1.17, 1.16, 1.15, and 1.11 in the build matrix

**Describe alternatives you've considered**

Roll the dice each release and hope that it works.

**What version of eunomia are you using?**

eunomia version: v0.1.1

**Additional context**

In the future we will want to make sure that we add a build matrix for OpenShift versions too. But that is not in scope for this enhancement request.

**Comment by seanmalloy** [comment]
Link to Travis CI build matrix documentation ... https://docs.travis-ci.com/user/build-matrix/

**Comment by seanmalloy** [comment]
The initial test matrix has been created and minikube has been updated to the latest version. I still need to ...

- finalize the k8s versions to be used in the test matrix
- fix issue with the deploy step running for each item in the matrix

**Comment by seanmalloy** [comment]
I need to do something similar to this to only run the deploy stage once.

https://docs.travis-ci.com/user/build-stages/matrix-expansion/

**Comment by seanmalloy** [comment]
It works ... https://travis-ci.com/KohlsTechnology/eunomia/builds/148846276.
